src/core/config/srp_client.h

```diff
@@ -70,6 +70,19 @@
 #define OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_MODE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_ANYCAST_SERVER_PORT
+ *
+ * Specifies the default port number to use to connect to a SRP server selected using anycast address from a related
+ * service entry in Thread Network Data.
+ *
+ * This config is applicable only when `OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE` is enabled.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_ANYCAST_SERVER_PORT
```

Comment:
Does it imply that the default port only applies to anycast?
Normally a server would use same port no matter the client is using RLOC or ALOC?

Reply:
Yes. The default applies to anycast model. In anycast model, we want to use a known/fixed port number (to keep network-data small). In unicast model, we continue to include the port number in network data (mainly to keep it compatible with current model used in Apple implementation). The server however can choose to use the same default port number in all cases.

In general, I think the SRP server on Thread mesh needs to be aware of how it wants to publish its info in network data, i.e. anycast vs unicast model (which then impacts the port it may use along with some other behaviors, e.g. SRP replication, etc.).

Reply:
Is SRP server going to bind on multiple ports? @wgtdkp 
If not, maybe let's just use same port for ALOC and other addresses?